posthog/hogql/functions/mapping.py

```diff
@@ -791,7 +791,7 @@ def compare_types(arg_types: list[ConstantType], sig_arg_types: tuple[ConstantTy
     "roundAge": HogQLFunctionMeta("roundAge", 1, 1),
     "roundDown": HogQLFunctionMeta("roundDown", 2, 2),
     # maps
-    "map": HogQLFunctionMeta("map", 0, 2),
+    "map": HogQLFunctionMeta("map", 0, None),
```

Comment:
Not something we should do now, but it'd be interesting if we could restrict at the HogQL level that we need an even number of parameters for the `map` function